### PR TITLE
Field rename

### PR DESCRIPTION
**Pull request by johnliucn925**
Add function to support field rename.

### CODE REVIEW

File: src/test/groovy/graphql/nadel/NadelTest.groovy [hunk 1]
```diff
@@ -64,19 +64,25 @@ class NadelTest extends Specification {
     def "stitching with service hydration"() {
         def dsl = """
         service FooService {
+            schema {
```
[the remark this answers is not in the document]
Comment (johnliucn925) [reply src/test/groovy/graphql/nadel/NadelTest.groovy]: It is not required for hydration , but it is required for field rename.

[the remark this answers is not in the document]
Comment (tsroka) [reply src/test/groovy/graphql/nadel/NadelTest.groovy]: there is a need for schema in field rename as they use schema to determine which type is query type. so we want to be consistent

[the remark this answers is not in the document]
Comment (andimarek) [reply src/test/groovy/graphql/nadel/NadelTest.groovy]: the graphql spec says that the schema definition is optional if the query object is named Query (and the optional mutation type is named Mutation).
If anybody needs this definition it is a bug.

File: src/test/groovy/graphql/nadel/NadelTest.groovy [hunk 2]
```diff
@@ -107,4 +113,38 @@ class NadelTest extends Specification {
...
+            service FooService {
+                schema {
+                    query: Query
+                }
+                type Query {
+                    foo: [Foo!]
+                }
+    
+                type Foo {
+                    newName: ID <= \$source.id
```
[the remark this answers is not in the document]
Comment (johnliucn925) [reply src/test/groovy/graphql/nadel/NadelTest.groovy]: Have separate ticket to verify invalid schema.

**Pull request merged**